assets/src/domain/eventEditor/ui/datetimes/datesList/actionsMenu/DateEntityMainMenuItem.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { __ } from '@wordpress/i18n';
+
+import { EspressoDropdownMenu } from '@application/ui/layout';
+import { Icon } from '@application/ui/input';
+import { useDatetimeContext } from '@edtrHooks/index';
+import { useDatetimeItem } from '@edtrServices/apollo/queries';
+import { useDatetimeMutator } from '@edtrServices/apollo/mutations';
+import { useFormModal } from '@appLayout/formModal';
+
+const DateEntityMainMenuItem = () => {
+	const { id } = useDatetimeContext();
+	const { dbId: entityDbId } = useDatetimeItem({ id }) || {};
+	const { openEditor } = useFormModal();
+	const { deleteEntity } = useDatetimeMutator(id);
+
+	const onEditClick = (): void =>
+		openEditor({
+			editorId: 'editDatetime',
+			entityId: id,
+			entityDbId,
+		});
+
+	return (
+		<EspressoDropdownMenu
+			icon={Icon.MORE}
+			label={__('event date main menu')}
+			controls={[
+				{
+					title: __('edit date'),
+					icon: Icon.EDIT,
+					onClick: onEditClick,
+					// onClick: useOpenEditor(useEventDateEditorId(dateEntity)),
+				},
+				// {
+				// 	title: __('copy date'),
+				// 	icon: 'admin-page',
+				// 	onClick: useCopyDateEntity(dateEntity),
+				// },
+				{
+					title: __('trash date'),
+					icon: 'trash',
+					onClick: () => deleteEntity({ id }),
+				},
+			]}
```

Comment:
i'm wondering if it might be better to change this and instead just pass a JSX element for each menu item instead of some configuration details

pros and cons to both approaches...
___
### Defining Menu Items in a Config Object (like above)
#### pros
- relatively simple to add new menu items
- all menu items should render the same because they would all use the same logic
- tabbing and a11y features can likely be better achieved for same reason as above
- tooltips would apply to the entire menu item in a consistent and reliable manner
- guaranteed safety and consistency with respects to what can be added to menu since items are defined solely by config object
#### cons
- not reusable in any way and potentially less DRY if a similar button is required elsewhere to perform the same thing (like delete a date) 
- violation of SRP - logic is not self contained, meaning the menu component also contains logic for handling events (like `deleteEntity()`)
- less versatile. the same thing that make this approach more consistent also makes it less able to handle a completely different use case
___
### Supplying Menu Items as JSX Elements
#### pros
- also relatively simple to add new menu items since any renderable node could be used
- far more reusable and DRY because the JSX elements passed to the menu could potentially be reused anywhere
- better adherence to SRP since the `DateEntityMainMenu` would simply aggregate other items for display
- far more versatile because nearly anything could be added to a menu (technically there would be no barriers)
- menu items could potentially still be controlled via TS and or using a wrapper to maintain consistency
#### cons
- menu items may not render the same because they could be anything
- tabbing and a11y features may be more difficult to achieve because of lack of control over menu items
- tooltips could also be troublesome if responsibility is left solely to the incoming component, and/or duplicated by the dropdown 

will need to think about this because I keep flip flopping back n forth thinking about which is better

what do you think?

Reply:
Yes, agree that this can be made more reusable.
Another approach we can take inspiration from is the antd select: https://ant.design/components/select/

Reply:
Yes, it is better like this:
```tsx
<Menu>
    <MenuItem>Label 1</MenuItem>
    <MenuItem>Label 2</MenuItem>
    <MenuItem>Label 3</MenuItem>
</Menu>
```
The API could expose both `Menu` and `MenuItem`